### DISCUSSION TITLE
Wiring ideas

### DISCUSSION THREAD

**Discussion by kleiner-sentry** [FakeFishGames/Barotrauma#11187] [Ideas and suggestions]
I know other people might have suggested this before, however i feel that it's important to make sure that we're heard, and that these suggestions get implemented at some point. and those suggestions are:
- An edge detector that will output the edge of a different number/letter (with an option to only output 1 when an edge is detected)
- add set_falseoutput to all components that has a false output in their config.

While it may reduce the cost of circuits, i feel it would not only allow us to simplify our circuits, but also make it easier to accomplish what we can already do.  Adding an edge detector would allow us to reliably convert a continuous signal into a single pulse, which has been an issue especially with latency in multiplayer. one may claim that it would be rarely used, and while that may be true, other components like acos and modulo are hardly ever used from my experience in wiring. Additionally, Adding a set_falseoutput would allow us to simplify a variety of circuits, and allow us to make them without having to do unnecessary tricks in order to get the intended output. I feel like these changes arent too demanding, and are relatively simple. Adding these suggestions would benefit the game more than it'd make worse, in my opinion. and I doubt anyone in the community would oppose this.

**Comment by NaturalBornCamper** [comment]
Came here to suggest the same thing after struggling so much to get the minimum value between the oxygen detectors in all my rooms.
If only the "greater component" had a pin "set_false_output", it would make life sooo much easier! I could compare 2 values and output the first value if `false`, or return the second value if `true`. Instead we got to do a pile of weird shenanigans and ugly connections XD

Same applies to "signal check component", "or", "and", etc. It would seem logic that you can set "false_output" since you can already set "true_output"?

I can't believe nobody asked this before 4 days ago, I was sure it was in the suggestions and just came to see how many people voted for it (and add my vote haha).

@kleiner-sentry Might I suggest you change the title to something more specific like "Wiring ideas - Edge dectector and set_falseoutput" instead of just "Wiring ideas"? It will be more descriptive and more people will find this on Google, Github search. Took me a while to find your suggestion and almost created a new one.

**Comment by mygamingaccount** [comment]
Doesn't it feel like they've made a decision not to create new gameplay elements related to the component network? I certainly don't expect anything to be added before 1.0 is released. Maybe after? Who knows.

The alternatives we have to a proper edge detector are all clunky. Guess we have to get used to them.
